### ISSUE TITLE
eth_feeHistory doesn't support block tags

### ISSUE THREAD

**Issue by spacesailor24** [ethereum/go-ethereum#23327]
#### System information

Geth version: `v1.10.6`

#### Expected behaviour

It is mentioned in the [RPC spec](https://playground.open-rpc.org/?uiSchema%5BappBar%5D%5Bui:splitView%5D=false&schemaUrl=https://raw.githubusercontent.com/ethereum/eth1.0-specs/master/json-rpc/spec.json&uiSchema%5BappBar%5D%5Bui:input%5D=false) that for `eth_feeHistory` block tags (e.g. `earliest`, `latest`, `pending`) are supported

#### Actual behaviour

An error is returned when attempting to use any block tag:

```json
{
  "jsonrpc": "2.0",
  "id": 0,
  "error": {
    "code": -32602,
    "message": "invalid argument 0: json: cannot unmarshal string into Go value of type int"
  }
}
```

And the [method implementation](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L90) seems to be expecting an `int`

#### Steps to reproduce the behaviour

```bash
curl -X POST --data '{"jsonrpc": "2.0","method": "eth_feeHistory","params": ["0x4","latest",[]],"id": 0}'
```

**Comment by ligi** [comment]
the problem is not the block tag - the problem is that the first parameter must be a number in 1.10.6 - it is fixed on master and will be rolled out with 1.10.7 that you can use hex strings here.
Currently you need to do:
`
curl -X POST --data '{"jsonrpc": "2.0","method": "eth_feeHistory","params": [4,"latest",[]],"id": 0}'
`

**Comment by spacesailor24** [comment]
> invalid argument 0

Whoops, that makes sense. Guess I'll close this and await `1.10.7`

Thank you!
